legacy/Makefile

```diff
@@ -14,7 +14,8 @@ endif
 
 
 hc-stream: ../main.cpp HCStream.cpp
```

Comment:
This file rename looks wrong. There might be multiple implementations placed in `legacy/` so we need to keep the Makefile name.